test/tstRay.cpp

```diff
@@ -571,12 +576,12 @@ BOOST_AUTO_TEST_CASE(ray_triangle_intersection,
   // between the origin of the ray and the vertices.
   // These tests will fail if there is no normalization.
   float const size_s = 0.0001;
-  Triangle small_triangle{{-size_s, 0, 0}, {0, size_s, 0}, {0, 0, size_s}};
+  Triangle small_triangle{Point{-size_s, 0.f, 0.f}, Point{0.f, size_s, 0.f}, Point{0.f, 0.f, size_s}};
```

Comment:
We really need to figure out how to construct a `Point` from integral types. This is annoying. And also prevents us from removing the original `Point`.

Reply:
If we always want `Point<dim, float>` in case template arguments are not explicitly specified, we could do something like https://godbolt.org/z/76Mx387q9.

Reply:
I think that almost does what we want. But we would want to cast it not to `float`, but something more complex:
- If all arguments can be promoted to `float`, use `float`
- If some arguments are `double`, use `double`.

Reply:
The alternative here would be to specify the template arguments for `Triangle` explicitly, see https://godbolt.org/z/oz1cTobTj.

Reply:
```C++
template <typename... T>
Dummy(T...)->Dummy<sizeof...(T), std::conditional_t<(... || std::is_same_v<std::decay_t<T>, double>), double, float>>;
```
would use `double` if any of the template arguments is `double` and `float` otherwise.

Reply:
Maybe that's what we want. Can you please try it with aprokop/ArborX#17?